pkg/bastion/acl.go

```diff
@@ -46,5 +50,51 @@ func checkACLs(user dbmodels.User, host dbmodels.Host) string {
 	}
 	sort.Sort(byWeight(acls))
 
-	return acls[0].Action
+	return checkACLsHook(aclCheckCmd, acls[0].Action, user, host)
+}
+
+// checkACLsHook executes external command to check ACL and passes following parameters:
+// $1 - SSH Portal action (`allow` or `deny`)
+// $2 - User as JSON string
+// $3 - Host as JSON string
+// External program has to return `allow` or `deny` in stdout.
+func checkACLsHook(aclCheckCmd string, action string, user dbmodels.User, host dbmodels.Host) string {
+	if aclCheckCmd != "" {
+		jsonUser, err := json.Marshal(user)
+		if err != nil {
+			log.Printf("Error: %v", err)
+			return action
+		}
+
+		jsonHost, err := json.Marshal(host)
+		if err != nil {
+			log.Printf("Error: %v", err)
```

Comment:
@GreyOBox maybe the log should be type of Errorf?

Reply:
sorry, didn't get what you mean. Could you add some example?

like that?
```
	err := fmt.Errorf("user %q (id %d) not found", name, id)
	fmt.Println(err.Error())
```
We'll not use `err` below, only print its value. So I guessed that `Printf` is ok here.

Reply:
Yap since we are not returning the actual error and we cannot wrap it with more context, we can format the error with Errorf before loging 
but let's see what will happen with the 1st comment :D

Reply:
If we do something to improve error handling, it will be by having typed errors and wrapping everywhere, I prefer to stay consistent with the rest of the codebase and keep the `log.Printf("Error: %v")` in that specific case